make/libraries

```diff
@@ -73,24 +78,36 @@ $(BOOST)/user-config.jam:
 $(BOOST_LIB)/mpi.so: $(BOOST)/user-config.jam
 	@mkdir -p $(dir $@)
 	cd $(BOOST); ./bootstrap.sh
-	cd $(BOOST); ./b2 --user-config=user-config.jam --layout=system --with-mpi --with-serialization -j$(BOOST_PARALLEL_BUILD) variant=release link=shared threading=multi runtime-link=shared
+	cd $(BOOST); ./b2 --user-config=user-config.jam --layout=system --with-mpi --with-serialization -j$(BOOST_PARALLEL_JOBS) variant=release link=shared threading=multi runtime-link=shared
 
 $(BOOST_LIB)/libboost_serialization.so: $(BOOST_LIB)/mpi.so
+$(BOOST_LIB)/libboost_mpi.so: $(BOOST_LIB)/mpi.so
+
 
 $(BOOST_LIB)/libboost_serialization.dylib: $(BOOST_LIB)/mpi.so
-	install_name_tool -add_rpath "$(BOOST_LIB_ABS)" "$(BOOST_LIB)/libboost_serialization.dylib"
+	install_name_tool -add_rpath "$(BOOST_LIBRARY_ABSOLUTE_PATH)" "$(BOOST_LIB)/libboost_serialization.dylib"
```

Comment:
You are fine with `BOOST_LIB`, but not with `BOOST_LIB_ABS`? I prefer the shorter version, but I definitely would like to see consistency with both variables names. So I would suggest `BOOST_LIB_ABS_PATH` or spell out `LIB` everywhere.

Reply:
Ah... I was going back and forth on even using `BOOST_LIB`. I had tried to remove it and then included it again while things were broken. I'll just remove. I want things to be explicit so there's less confusion. (I had a hell of a time understanding the difference between `BOOST` and `BOOST_LIB` and `BOOST_LIB_ABS`. That alone took me half an hour to pick apart the intent. I want to avoid that in the future. If it means names are a little verbose, so be it.)

Anyway, I'll remove `BOOST_LIB`. For a while, I was thinking you wanted to support a binary installation of Boost, but I realized that our make doesn't allow it, so I'll remove the option to.